src/libraries/Microsoft.Extensions.Caching.Abstractions/ref/Microsoft.Extensions.Caching.Abstractions.csproj

```diff
@@ -6,6 +6,11 @@
 
   <ItemGroup>
     <Compile Include="Microsoft.Extensions.Caching.Abstractions.cs" />
+    <Compile Include="Microsoft.Extensions.Caching.Abstractions.other.cs" Condition="'$(TargetFramework)' != 'net7.0'" />
+    <Compile Include="Microsoft.Extensions.Caching.Abstractions.net7.cs" Condition="'$(TargetFramework)' == 'net7.0'" />
```

Comment:
Does this play nicely with `t:GenerateReferenceAssemblySource`? Can we do `#if NET7_0_OR_GREATER` in the reference assembly source so we don't have to change this when we start on `net8.0`?

Reply:
I don't see any precedence with using ifdef with ref files. Perhaps the condition here could be improved to also cover not just net7.0 but also higher. Also, I assume it won't be possible to use `\t:GenerateReferenceAssemblySource` alongside this.

/cc @joperezr

Reply:
I think Stephen Halter idea will work too.

Reply:
Just to confirm, I tried using the `GenerateReferenceAssemblySource` and noticed it would add `GetCurrentStatistics` to IMemoryCache on `ref\Microsoft.Extensions.Caching.Abstractions.cs`. That's not exactly what we would want, since we prefer to move this addition to a separate `*.net7.cs` file and include it conditionally:

```diff
    public partial interface IMemoryCache : System.IDisposable
    {
        Microsoft.Extensions.Caching.Memory.ICacheEntry CreateEntry(object key);
+        Microsoft.Extensions.Caching.Memory.MemoryCacheStatistics? GetCurrentStatistics() { throw null; }
        void Remove(object key);
        bool TryGetValue(object key, out object? value);
    }
```